.github/workflows/module-validation.yml

```diff
@@ -0,0 +1,36 @@
+name: Validate module imports
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+jobs:
+  tach:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.12.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: pip install tach==0.13.1
+
+      - uses: actions/checkout@v4

```

Comment:
Let's make this the first step ... otherwise it can be weird (the clone wipes the github workspace)

Reply:
Done